python3Packages.nocturne: init at unstable-2022-10-15

samuela opened the pull request:
###### Description of changes

1. Add the nocturne simulator
2. Fix https://github.com/NixOS/nixpkgs/issues/208843
3. Alias `python3Packages.virtual-display` to `python3Packages.PyVirtualDisplay`, the package's real name

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

samuela committed:
python3Packages.hydra-core: fix #208843

wegank commented on pkgs/development/python-modules/nocturne/default.nix:
```diff
@@ -0,0 +1,61 @@
+
+buildPythonPackage rec {
+  pname = "nocturne";
+  version = "unstable-2023-01-02";
```
https://github.com/facebookresearch/nocturne/commit/ae0a4e361457caf6b7e397675cc86f46161405ed

```suggestion
  version = "unstable-2022-10-15";
```

samuela replied:
good point!

samuela commented:
if no one objects I'll go ahead and merge tomorrow

bcdarwin commented:
@ofborg build python310Packages.nocturne

bcdarwin commented:
@ofborg succeeds, but `nixpkgs-review` fails to build this since it disallows the use of aliases, so probably this package should eschew them as well.

samuela committed:
python3Packages: virtual-display -> PyVirtualDisplay

virtual-display is a quite old package, and it's not clear why it was named "virtual-display" as the PyPI name is (and was) "PyVirtualDisplay". In any case, this commit rectifies that inconsistency.

samuela committed:
python3Packages.nocturne: init at unstable-2022-10-15

samuela commented:
@bcdarwin Good point, fixed in https://github.com/NixOS/nixpkgs/pull/208847/commits/ecf05f1c15eb348bf3f72cb2bc2729f6f10974ec

samuela commented on pkgs/development/python-modules/nocturne/dependencies.patch:
```diff
@@ -0,0 +1,27 @@
+     #     requires = f.read().splitlines()
+     setup(
++        # Adding `packages=` is necessary to get `python setup.py install` to work
++        packages=["nocturne", "cfgs"],
```
> I didn't check locally since nixpkgs-review is broken

How is it broken? I thought this was fixed after https://github.com/NixOS/nixpkgs/commit/ecf05f1c15eb348bf3f72cb2bc2729f6f10974ec

> does this cause a top-level importable cfgs package to be created?

Unfortunately, yes. But that's how nocturne is currently designed for better or worse. Even `import nocturne` doesn't work without having the top-level `cfgs` module present.

samuela commented on pkgs/development/python-modules/nocturne/default.nix:
```diff
@@ -0,0 +1,61 @@
+buildPythonPackage rec {
+  pname = "nocturne";
+  version = "unstable-2022-10-15";
+  format = "setuptools";
```
AFAICT nocturne does not specify a minimum python version, so I'd like to keep this unspecified for now. We can always add a `disabled = pythonOlder "..."` if version issues come up in the future

bcdarwin commented:
Sorry, forgot to publish this review earlier. Fixed now.

I guess it’s fine, but we should file a bug upstream.

On Sunday, 8 January 2023, Samuel Ainsworth ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In pkgs/development/python-modules/nocturne/dependencies.patch
> <https://github.com/NixOS/nixpkgs/pull/208847#discussion_r1064196053>:
>
> > +-  GIT_REPOSITORY https://github.com/google/googletest.git
> +-  GIT_TAG main
> ++  SOURCE_DIR @gtest_src@
> + )
> +
> + set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
> +diff --git a/setup.py b/setup.py
> +index 4863ae6..b5d3545 100644
> +--- a/setup.py
> ++++ b/setup.py
> +@@ -92,6 +92,8 @@ def main():
> +     # with open("./requirements.txt", "r") as f:
> +     #     requires = f.read().splitlines()
> +     setup(
> ++        # Adding `packages=` is necessary to get `python setup.py install` to work
> ++        packages=["nocturne", "cfgs"],
>
> I didn't check locally since nixpkgs-review is broken
>
> How is it broken? I thought this was fixed after ecf05f1
> <https://github.com/NixOS/nixpkgs/commit/ecf05f1c15eb348bf3f72cb2bc2729f6f10974ec>
>
> does this cause a top-level importable cfgs package to be created?
>
> Unfortunately, yes. But that's how nocturne is currently designed for
> better or worse. Even import nocturne doesn't work without having the
> top-level cfgs module present.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NixOS/nixpkgs/pull/208847#discussion_r1064196053>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABICNHP7CF3IKIOMJUHD3LWRMNXHANCNFSM6AAAAAATPLCIQM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

bcdarwin commented:
Makes sense.

On Sunday, 8 January 2023, Samuel Ainsworth ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In pkgs/development/python-modules/nocturne/default.nix
> <https://github.com/NixOS/nixpkgs/pull/208847#discussion_r1064196403>:
>
> > +, cmake
> +, fetchFromGitHub
> +, gtest
> +, hydra-core
> +, lib
> +, nlohmann_json
> +, pybind11
> +, PyVirtualDisplay
> +, sfml
> +, substituteAll
> +}:
> +
> +buildPythonPackage rec {
> +  pname = "nocturne";
> +  version = "unstable-2022-10-15";
> +  format = "setuptools";
>
> AFAICT nocturne does not specify a minimum python version, so I'd like to
> keep this unspecified for now. We can always add a disabled = pythonOlder
> "..." if version issues come up in the future
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NixOS/nixpkgs/pull/208847#discussion_r1064196403>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABICNFYEGNWZR734ZPZVMLWRMODLANCNFSM6AAAAAATPLCIQM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

samuela commented:
> I guess it’s fine, but we should file a bug upstream.

Yup, done in https://github.com/facebookresearch/nocturne/issues/63!
